src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -858,4 +859,30 @@ public static Map<String, String> getJobTags(String jobType) {
     labels.put("type", jobType);
     return labels;
   }
+
+  /**
+   * Identify a stating bucket name from the pipeline context and plugin configuration
+   * @param arguments runtime arguments
+   * @param configLocation location from plugin configuration
+   * @param dataset BigQuery dataset
+   * @param bucket bucket from plugin configuration
+   * @return Bucket name to use for this sink.
+   */
+  public static String getStagingBucketName(Map<String, String> arguments, @Nullable String configLocation,
+                                            @Nullable Dataset dataset, @Nullable String bucket) {
+    // Get Bucket Prefix from configuration
+    String bucketPrefix = BigQueryUtil.getBucketPrefix(arguments);
+
+    // If temp bucket name is not defined in configuration, and a common bucket name prefix has been specified,
+    // we must set this prefix along with the destination location in order to create/reuse the bucket.
+    // Otherwise, if temp bucket name is defined, or a prefix is not set, the configureBucket method will prepare
+    // for a new bucket creation.
+    if (Strings.isNullOrEmpty(bucket) && bucketPrefix != null) {
+      // If the destination dataset exists, use the dataset location. Otherwise, use location from configuration.
+      String datasetLocation = dataset != null ? dataset.getLocation() : configLocation;
+      // Get the bucket name for the specified location.
+      bucket = BigQueryUtil.getBucketNameForLocation(bucketPrefix, datasetLocation);
+    }
+    return bucket;
```

Comment:
How does this behave is the supplied Bucket is nullable and there is no prefix set? As far as I can tell, the returned `bucket` will be null.

If this is the case, please add the `@Nullable` annotation to this function.

Following question is: Should this function return a bucket name if not supplied and no prefix is set?

Reply:
Yeah, it is expected to return `null` in that case. Added the `@Nullable` annotation.

No we don't expect this method to return a bucket name if not supplied and if no prefix is set.
`BigQuerySinkUtils#configureBucket` and `BigQuerySourceUtils#getOrCreateBucket` expect the `bucket` argument to be `nullable` and keep the existing behaviour.